openff/interchange/_tests/unit_tests/test_models.py

```diff
@@ -202,6 +203,35 @@ def test_torsionkey_eq_hash():
     )
 
 
+def test_base_virtual_site_eq():
+    key1 = BaseVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="A", name="B")
+    key2 = BaseVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="F", name="B")
+    key3 = BaseVirtualSiteKey(orientation_atom_indices=[3, 2, 1], type="A", name="B")
+    key4 = BaseVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="A", name="B")
+
+    for key in [key1, key2, key3, key4]:
+        assert key == key
+
+    assert key1 != key2
+    assert key1 != key3
+    assert key2 != key3
+    assert key1 == key4
+
+
+def test_smirnoff_virtual_site_eq():
+    """This class is the same as BaseVirtualSiteKey but adds the `match` field."""
+    key1 = SMIRNOFFVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="A", name="B", match="once")
+    key2 = SMIRNOFFVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="A", name="B", match="all_permutations")
+    key3 = SMIRNOFFVirtualSiteKey(orientation_atom_indices=[1, 2, 3], type="A", name="B", match="once")
+
+    for key in [key1, key2, key3]:
+        assert key == key
+
+    assert key1 != key2
+    assert key1 == key3
+    assert key2 != key3
+
```

Comment:
Also test that hashes are equal to be consistent with `test_bondkey_eq_hash`
```suggestion
assert hash(key1) == hash(key3)
```